app/cni/cmd/kuma-cni/main.go

```diff
@@ -0,0 +1,266 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net"
+	"time"
+
+	"github.com/containernetworking/cni/pkg/skel"
+	"github.com/containernetworking/cni/pkg/types"
+	"github.com/containernetworking/cni/pkg/types/current"
+	"github.com/containernetworking/cni/pkg/version"
+
+	"github.com/kumahq/kuma/pkg/core"
+)
+
+const (
+	podRetrievalMaxRetries = 30
+	podRetrievalInterval   = 1 * time.Second
+)
+
+var (
+	log = core.NewLoggerWithRotation(2, "/tmp/kuma-cni.log", 100, 0, 0).WithName("kuma-cni")
+)
+
+// Kubernetes a K8s specific struct to hold config
+type Kubernetes struct {
+	Kubeconfig        string   `json:"kubeconfig"`
+	ExcludeNamespaces []string `json:"exclude_namespaces"`
+	CniBinDir         string   `json:"cni_bin_dir"`
+}
+
+type PluginConf struct {
+	types.NetConf // You may wish to not nest this type
+
+	RawPrevResult *map[string]interface{} `json:"prevResult"`
+	PrevResult    *current.Result         `json:"-"`
+
+	// plugin-specific fields
+	LogLevel   string     `json:"log_level"`
+	Kubernetes Kubernetes `json:"kubernetes"`
+}
+
+// K8sArgs is the valid CNI_ARGS used for Kubernetes
+// The field names need to match exact keys in kubelet args for unmarshalling
+type K8sArgs struct {
+	types.CommonArgs
+	IP                         net.IP
+	K8S_POD_NAME               types.UnmarshallableString // nolint: golint, stylecheck
+	K8S_POD_NAMESPACE          types.UnmarshallableString // nolint: golint, stylecheck
+	K8S_POD_INFRA_CONTAINER_ID types.UnmarshallableString // nolint: golint, stylecheck
+}
+
+// parseConfig parses the supplied configuration (and prevResult) from stdin.
+func parseConfig(stdin []byte) (*PluginConf, error) {
+	conf := PluginConf{}
+
+	if err := json.Unmarshal(stdin, &conf); err != nil {
+		return nil, fmt.Errorf("failed to parse network configuration: %v", err)
+	}
+
+	// Parse previous result. Remove this if your plugin is not chained.
+	if conf.RawPrevResult != nil {
+		resultBytes, err := json.Marshal(conf.RawPrevResult)
+		if err != nil {
+			return nil, fmt.Errorf("could not serialize prevResult: %v", err)
+		}
+		res, err := version.NewResult(conf.CNIVersion, resultBytes)
+		if err != nil {
+			return nil, fmt.Errorf("could not parse prevResult: %v", err)
+		}
+		conf.RawPrevResult = nil
+		conf.PrevResult, err = current.NewResultFromResult(res)
+		if err != nil {
+			return nil, fmt.Errorf("could not convert result to current version: %v", err)
+		}
+	}
+	// End previous result parsing
+
+	return &conf, nil
+}
+
+// cmdAdd is called for ADD requests
+func cmdAdd(args *skel.CmdArgs) error {
+	ctx := context.Background()
+	conf, err := parseConfig(args.StdinData)
+	if err != nil {
+		log.Error(err, "error parsing kuma-cni cmdAdd config")
+		return err
+	}
+	logPrevResult(conf)
+
+	// Determine if running under k8s by checking the CNI args
+	k8sArgs := K8sArgs{}
+	if err := types.LoadArgs(args.Args, &k8sArgs); err != nil {
+		log.Error(err, "error loading kuma-cni cmdAdd args")
+		return err
+	}
```

Comment:
imho here create 
```
logger := log.WithValues("pod", string(k8sArgs.K8S_POD_NAME), "namespace", string(k8sArgs.K8S_POD_NAMESPACE))
```
and pass this around instead of k8sargs

Reply:
I created a logger with a couple fields and used it in the whole cmdAdd